notebooks/nessie-iceberg-demo-nba.ipynb

```diff
@@ -0,0 +1,629 @@
+    "-------------------------------------\n",
+    "Once we created the `dev` branch and verified that it exists, we can create some tables and add some data.\n",
+    "\n",
+    "We create two tables under the `dev` branch:\n",
+    "- `salaries`\n",
+    "- `totals_stats`\n",
+    "\n",
+    "These tables list the salaries per player per year and their stats per year.\n",
+    "\n",
+    "To create the data we:\n",
+    "\n",
+    "1. switch our branch context to dev\n",
+    "2. create the table\n",
+    "3. insert the data from an existing csv file. This csv file is already stored locally on the demo machine. A production use case would likely take feeds from official data sources"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spark.sql(\"USE REFERENCE dev IN dev_catalog\")\n",
+    "\n",
+    "# Creating `salaries` table\n",
+    "spark.sql(\"CREATE TABLE IF NOT EXISTS dev_catalog.nba.salaries (Season STRING, Team STRING, Salary STRING, Player STRING) USING iceberg\")\n",
+    "\n",
+    "spark.sql('CREATE OR REPLACE TEMPORARY VIEW salaries_table USING csv OPTIONS (path \"../datasets/nba/salaries.csv\", header true)')\n",
+    "spark.sql('INSERT INTO dev_catalog.nba.salaries SELECT * FROM salaries_table')\n",
+    "\n",
+    "# Creating `totals_stats` table\n",
+    "spark.sql(\"\"\"CREATE TABLE IF NOT EXISTS dev_catalog.nba.totals_stats (\n",
+    "             Season STRING, Age STRING, Team STRING, ORB STRING, DRB STRING, TRB STRING, AST STRING, STL STRING, \n",
+    "             BLK STRING, TOV STRING, PTS STRING, Player STRING, RSorPO STRING) \n",
+    "             USING iceberg\"\"\")\n",
+    "spark.sql('CREATE OR REPLACE TEMPORARY VIEW stats_table USING csv OPTIONS (path \"../datasets/nba/totals_stats.csv\", header true)')\n",
+    "spark.sql('INSERT INTO dev_catalog.nba.totals_stats SELECT * FROM stats_table').toPandas()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we count the rows in our tables to ensure they are the same number as the csv files. Note we use the `table@branch` notation which overrides the context set by a `USE REFERENCE` command."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    },
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "table_count = spark.sql(\"select count(*) from dev_catalog.nba.`salaries@dev`\").toPandas().values[0][0]\n",
+    "csv_count = spark.sql(\"select count(*) from salaries_table\").toPandas().values[0][0]\n",
+    "assert table_count == csv_count\n",
+    "\n",
+    "table_count = spark.sql(\"select count(*) from dev_catalog.nba.`totals_stats@dev`\").toPandas().values[0][0]\n",
+    "csv_count = spark.sql(\"select count(*) from stats_table\").toPandas().values[0][0]\n",
+    "assert table_count == csv_count"
```

Comment:
Same here: Maybe add some `print` of these numbers here, so people have something to look at.

Reply:
good idea, done